### PR TITLE
Add NUMBER support

### PR DESCRIPTION
**Pull request by lpoulain**
Add NUMBER support

## Description

Support Trino NUMBER type by performing the following:

- Passes the NUMBER to the Trino Client Capabilities header so that such type is not returned as a varchar
- Maps this type to a Decimal

## Non-technical explanation

Support Trino NUMBER type

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(X) Release notes are required, with the following suggested text:

Support Trino NUMBER type

```markdown
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: tests/unit/test_mapper.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    @pytest.mark.parametrize("raw_type", ["number", "decimal"])
+    def test_number_and_decimal_use_decimal_mapper(self, raw_type):
+        mapper = RowMapperFactory()._create_value_mapper(_column(raw_type))
+        assert isinstance(mapper, DecimalValueMapper)
+
+    def test_number_column_maps_value_to_decimal(self):
+        mapper = RowMapperFactory()._create_value_mapper(_column("number"))
+        result = mapper.map("10.3")
+        assert isinstance(result, Decimal)
+        assert result == Decimal("10.3")
```
Comment (azawlocki-sbdt) [inline tests/unit/test_mapper.py:45]: Please add test cases for exponential notation (positive and negative exponents) and negative literals.

**Review by azawlocki-sbdt — CHANGES_REQUESTED** [review]
I'd add some more tests for numeric literals, otherwise LGTM

**Comment by wendigo** [comment]
cc @findepi

File: tests/unit/test_mapper.py (hunk 1 above)
Comment (lpoulain) [reply tests/unit/test_mapper.py:45]: I've added some extra tests, including the NUMBER corner cases.

**Review by wendigo — APPROVED** [review]

File: tests/unit/test_mapper.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        mapper = RowMapperFactory()._create_value_mapper(_column("number"))
+        assert mapper.map("-2.5E-2") == Decimal("-0.025")
+
+    def test_number_column_maps_none(self):
+        mapper = RowMapperFactory()._create_value_mapper(_column("number"))
+        assert mapper.map(None) is None
+
+    def test_number_column_maps_infinity(self):
+        mapper = RowMapperFactory()._create_value_mapper(_column("number"))
+        assert mapper.map("Infinity") == Decimal("Infinity")
```
Comment (findepi) [inline tests/unit/test_mapper.py]: Trino wire protocol uses "+Infinity" rather than "Infinity"

https://github.com/trinodb/trino/blob/873243431f962de6d7a51d3a75214f5fc0c58517/client/trino-jdbc/src/main/java/io/trino/jdbc/AbstractTrinoResultSet.java#L163-L168

**Review by findepi — APPROVED** [review]
add test with `+Infinity`

File: tests/unit/test_mapper.py (hunk 2 above)
Comment (lpoulain) [reply tests/unit/test_mapper.py]: Good catch. Fixed.

**Review by azawlocki-sbdt — APPROVED** [review]
LGTM

**Review by wendigo — APPROVED** [review]
